### PR TITLE
Provide utility to overlap table cell padding

### PR DESCRIPTION
**Pull request by lyubomir-popov**
## Done

Provide utility to overlap table cell padding

## QA

- Pull code
- Run `./run serve --watch`
- Open examples/utilities/table-row-padding-overlap/, verify example is correct and classes work as expected

## Details

Fixes https://github.com/canonical-web-and-design/vanilla-framework/issues/2504
Fixes https://github.com/canonical-web-and-design/vanilla-framework/issues/2463

### CODE REVIEW

File: docs/examples/utilities/table-cell-padding-overlap.html [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+        <td><button class="u-table-cell-padding-overlap">Select</button></td>
+        <td>Text</td>
+    </tr>
+    <tr>
+      <td><button class="is-dense u-table-cell-padding-overlap">Select</button></td>
+      <td>Text not wrapped in a tag</td>
+    </tr>
+    <tr>
+      <td><button class="u-table-cell-padding-overlap">Select</button></td>
+      <td><p class='u-no-margin--bottom u-table-cell-padding-overlap'>text in a paragraph</p></td>
```
Comment (kwm14) [inline docs/examples/utilities/table-cell-padding-overlap.html]: Capitalise 'Text' so it's consistent with other table row examples

**Comment by kwm14** [comment]
Design LGTM 👍 just once comment for QA above 😄

**Comment by lyubomir-popov** [comment]
@kwm14 well spotted, thanks! Fixed.

File: scss/_settings_spacing.scss [hunk 2]
```diff
@@ -147,6 +147,9 @@ $icon-sizes: (
   thumb--large: $sp-unit * 12
 ) !default;
 
+$table-cell-padding-top: $spv-inner--x-small--scaleable !default;
+$table-cell-padding-bottom: $spv-inner--x-small--scaleable !default;
```
Comment (anthonydillon) [inline scss/_settings_spacing.scss]: Wouldn't we want these to always be the same value? If these are override able (!default) then they should be documented?

Comment (lyubomir-popov) [reply scss/_settings_spacing.scss]: Good point. The only purpose of these is to ensure table cell padding and the overlap utility always match, so no need for `!default` and no need for separate top/bottom vars. I've updated the code.

**Comment by lyubomir-popov** [comment]
This is ready for another look.

**Pull request merged**